### PR TITLE
sparcv8leon3: add DMA support to uart driver

### PR DESCRIPTION
**Pull request by lukileczo**
## Motivation and Context

Add DMA to UART driver.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: `sparcv8leon3-gr716-mimas`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work https://github.com/phoenix-rtos/phoenix-rtos-project/pull/1018, https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/523
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: dma/grdmac2/grdmac2.c [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+#include <sys/mman.h>
+#include <sys/platform.h>
+
+#if defined(__CPU_GR716)
+#include <phoenix/arch/sparcv8leon3/gr716/gr716.h>
+#elif defined(__CPU_GR712RC)
+#include <phoenix/arch/sparcv8leon3/gr712rc/gr712rc.h>
+#else
+#error "Unsupported target"
+#endif
```
Comment (agkaminski) [inline dma/grdmac2/grdmac2.c:26]: Would be nice to move this to some common header file, as it's needed in multiple .c files in this driver

File: dma/grdmac2/grdmac2.c [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	return EOK;
+}
+
+
+void grdma_destroy(grdma_ctx_t *ctx)
+{
+	if ((ctx != NULL) && (ctx->descr != NULL)) {
+		(void)munmap(ctx->descr, ctx->descrSize);
+	}
+	free(ctx);
```
Comment (agkaminski) [inline dma/grdmac2/grdmac2.c:97]: `ctx->base` is "leaked"

File: multi/grlib-multi/uart.c [hunk 3]
```diff
@@ -49,6 +60,13 @@
 #define UART_SCALER 3
 #define UART_DEBUG  4
 
+/* Byte offset of LSB of UART_DATA register */
+#if defined(__BIG_ENDIAN)
```
Comment (agkaminski) [inline multi/grlib-multi/uart.c:64]: `__BIG_ENDIAN` is always defined
```suggestion
#if __BYTE_ORDER == __BIG_ENDIAN
```

**Review by maska989 — APPROVED** [review]
Tested on Experimental target sparcv8leon3-gr716-mimas everything works like a charm LGTM

**Review by agkaminski — APPROVED** [review]

**Pull request merged**